src/flexmeasures_client/client.py

```diff
@@ -52,27 +56,40 @@ async def request(
         if self.session is None:
             self.session = ClientSession()
 
-        try:
-            async with async_timeout.timeout(self.request_timeout):
-                response = await self.session.request(
-                    url=url,
-                    method=method,
-                    params=params,
-                    headers=headers,
-                    json=json,
-                    ssl=False,
-                )
-                response.raise_for_status()
-        except asyncio.TimeoutError as exception:
-            msg = "Timeout occurred while connecting to the API."
-            raise ConnectionError(
-                msg,
-            ) from exception
-        except (ClientError, socket.gaierror) as exception:
-            msg = "Error occurred while communicating with the API."
-            raise ConnectionError(
-                msg,
-            ) from exception
+        retry_function = lambda x, y: getattr(x, "status") == 400 and (
```

Comment:
Better name: `is_scheduling_job_done`.

We can also rename `x` to `exc` and `y` to `pl`, so it becomes a bit readable.